### DISCUSSION TITLE
ESP Main.c missing and cummins info

### DISCUSSION THREAD

**Discussion by Decryptortuning** [laszlodaniel/ChryslerScanner#15] [General]
Hello,
@laszlodaniel 
Stumbled upon your project. I saw links to source via tindie, came here, and found bins. Is this repo only for releases? I dug around and found main.c for the ESP as expected but in some later commit it goes away. 
Backround: Built esp32 based generic datalink tool for random one-off needs. I am implementing VPW via the RMT driver and wanted to look at your implementation and ask how your timings ended up working out with freertos's interrupt overheads.
Fun fact, I come from a long time background of reverse engineering ECM's and vehicle technology. I stumbled upon some of your Cummins support. I have some documentation that would make your life a lot easier and give some major help. Hoping I can collaborate with you some.
![image](https://github.com/laszlodaniel/ChryslerScanner/assets/19156562/52534007-3c0a-457a-97d7-7e4fd223e4fb)

**Comment by Decryptortuning** [comment]
Downloaded the commit from last june that seemed complete. Looks like your using external GPIO expander via I2C? May I ask what your timings are like? Are you able to keep strict control? I  found the freertos overhead extremely challenging on getting timings correct with the internal RMT peripheral, let alone freertos I2C talking to external hardware.

**Comment by laszlodaniel** [comment]
Hi @Decryptortuning , that main.c is long obsolete. Things have changed in recent years and I stopped sharing source code.

VPW is handled by hardware timers and GPIO interrupts; RMT is not used at all.
There are no timing issues, even though a bunch of FreeRTOS tasks are running with queues blocking them.

The I2C port expander is not causing any trouble.

I have yet to test active BLE communication if it has any effect.

Cummins docs: maybe some other time, thanks for offering!
